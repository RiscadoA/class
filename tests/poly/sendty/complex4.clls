proc main() {
    cut {
        println("sendty"); sendty x(lint);
        println("lhs send"); send x(y. println("let y"); let y 7);
        println("lhs recv"); recv x(y);
        println("wait"); wait x;
        println(y); ()
        |x: recvty T; recv ~T; send T; close|
        println("recvty"); recvty x(U);
        println("rhs recv"); recv x(y);
        println("rhs send"); send x(z.
            cut {
                println("fwd 1"); fwd y w
                |w: ~U|
                println("fwd 2"); fwd w z
            }
        );
        println("close"); close x
    }
};;
