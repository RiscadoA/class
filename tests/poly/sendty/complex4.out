sendty
lhs send
let y
recvty
rhs recv
rhs send
close
fwd 1
fwd 2
lhs recv
wait
7
